src/shop-app.html

```diff
@@ -18,11 +18,25 @@
 <link rel="import" href="../bower_components/iron-media-query/iron-media-query.html">
 <link rel="import" href="../bower_components/iron-pages/iron-pages.html">
 <link rel="import" href="../bower_components/iron-selector/iron-selector.html">
+<link rel="import" href="../bower_components/paper-icon-button/paper-icon-button.html">
 <link rel="import" href="shop-category-data.html">
 <!-- Since 'home' is the default route, eagerly load it. -->
 <link rel="import" href="shop-home.html">
 
+
+
 <dom-module id="shop-app">
+  <!--
+    TODO(ricitc): make these lazy-imports after
+        https://github.com/Polymer/polymer-analyzer/pull/543 lands
+        DO NOT MERGE until then :)
+  -->
+  <link rel="import" href="shop-cart-data.html">
+  <link rel="import" href="lazy-resources.html">
+  <link rel="import" href="shop-list.html">
+  <link rel="import" href="shop-detail.html">
+  <link rel="import" href="shop-cart.html">
+  <link rel="import" href="shop-checkout.html">
```

Comment:
I have voiced my opinion in the #tools channel and in direct messages with you on Slack, that I really dislike this pattern. It essentially requires users to 1. specify the routes in the selector, 2. specify the fragments and 3. specify the lazy-imports. This is not DRY and is very error-prone for the users.

Point 2 of the above list can of course be derived from point 3, but why can't these both be derived from point 1? The problem specifically in the shop app is that it relies on `Polymer.Base.importHref` which is extremely hard to analyze statically. Therefore I would like to request a solution that is statically analyzable, which is DRY (in the sense that it only requires users to specify the routes in the selector) and user-friendly.

Any sort of hint or pattern for the analyzer to conclude the different fragments that are lazy-loaded would be sufficient, to which it can combine the final document locations together with the routes. I am not aware of any clear solution, which would work for all users, given that the shop lazy-loading pattern is very specific to this application. A lot of other users are probably relying on a different mechanism, but at the moment I do not know all of the patterns that are used.